internal/loadbalancer/data_source_test.go

```diff
@@ -0,0 +1,71 @@
+package loadbalancer_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/terraform-providers/terraform-provider-hcloud/internal/loadbalancer"
+	"github.com/terraform-providers/terraform-provider-hcloud/internal/testsupport"
+	"github.com/terraform-providers/terraform-provider-hcloud/internal/testtemplate"
+)
+
+func TestAccHcloudDataSourceLoadBalancerTest(t *testing.T) {
+	tmplMan := testtemplate.Manager{}
+
+	res := &loadbalancer.RData{
+		Name:         "some-load-balancer",
+		LocationName: "nbg1",
+		Labels: map[string]string{
+			"key": "value",
+		},
+	}
+	lbByName := &loadbalancer.DData{
+		Name:             "lb_by_name",
+		LoadBalancerName: res.HCLID() + ".name",
+	}
+	lbByID := &loadbalancer.DData{
+		Name:           "lb_by_id",
+		LoadBalancerID: res.HCLID() + ".id",
+	}
+	lbBySel := &loadbalancer.DData{
+		Name:          "lb_by_sel",
+		LabelSelector: fmt.Sprintf("key=${%s.labels[\"key\"]}", res.HCLID()),
```

Comment:
I personally thing `HCLID` in this context a bit confusing. On the first look I read "Hetzner Cloud ID" and was a bit confused about this. Maybe rename the function to something like `TFID` or `TerraformID`?